crates/spirv-std/src/arch/atomic.rs

```diff
@@ -0,0 +1,487 @@
+    ptr: &mut U,
+    value: U,
+) -> U {
+    let mut old = U::default();
+
+    asm! {
+        "%u32 = OpTypeInt 32 0",
+        "%scope = OpConstant %u32 {scope}",
+        "%semantics = OpConstant %u32 {semantics}",
+        "%value = OpLoad _ {value}",
+        "%old = OpAtomicUMax _ {ptr} %scope %semantics %value",
+        "OpStore {old} %old",
+        scope = const SCOPE,
+        semantics = const SEMANTICS,
+        ptr = in(reg) ptr,
+        old = in(reg) &mut old,
+        value = in(reg) &value
+    }
+
+    old
+}
+
+/// Atomically and an integer and return the old value.
+#[spirv_std_macros::gpu_only]
+#[doc(alias = "OpAtomicAnd")]
+#[inline]
+pub unsafe fn atomic_and<I: Integer, const SCOPE: u32, const SEMANTICS: u32>(
+    ptr: &mut I,
+    value: I,
+) -> I {
+    let mut old = I::default();
+
+    asm! {
+        "%u32 = OpTypeInt 32 0",
+        "%scope = OpConstant %u32 {scope}",
+        "%semantics = OpConstant %u32 {semantics}",
+        "%value = OpLoad _ {value}",
+        "%old = OpAtomicAnd _ {ptr} %scope %semantics %value",
+        "OpStore {old} %old",
+        scope = const SCOPE,
+        semantics = const SEMANTICS,
+        ptr = in(reg) ptr,
+        old = in(reg) &mut old,
+        value = in(reg) &value
+    }
+
+    old
+}
+
+/// Atomically or an integer and return the old value.
+#[spirv_std_macros::gpu_only]
+#[doc(alias = "OpAtomicOr")]
+#[inline]
+pub unsafe fn atomic_or<I: Integer, const SCOPE: u32, const SEMANTICS: u32>(
+    ptr: &mut I,
+    value: I,
+) -> I {
+    let mut old = I::default();
+
+    asm! {
+        "%u32 = OpTypeInt 32 0",
+        "%scope = OpConstant %u32 {scope}",
+        "%semantics = OpConstant %u32 {semantics}",
+        "%value = OpLoad _ {value}",
+        "%old = OpAtomicOr _ {ptr} %scope %semantics %value",
+        "OpStore {old} %old",
+        scope = const SCOPE,
+        semantics = const SEMANTICS,
+        ptr = in(reg) ptr,
+        old = in(reg) &mut old,
+        value = in(reg) &value
+    }
+
+    old
+}
+
+/// Atomically xor an integer and return the old value.
+#[spirv_std_macros::gpu_only]
+#[doc(alias = "OpAtomicXor")]
+#[inline]
+pub unsafe fn atomic_xor<I: Integer, const SCOPE: u32, const SEMANTICS: u32>(
+    ptr: &mut I,
+    value: I,
+) -> I {
+    let mut old = I::default();
+
+    asm! {
+        "%u32 = OpTypeInt 32 0",
+        "%scope = OpConstant %u32 {scope}",
+        "%semantics = OpConstant %u32 {semantics}",
+        "%value = OpLoad _ {value}",
+        "%old = OpAtomicXor _ {ptr} %scope %semantics %value",
+        "OpStore {old} %old",
+        scope = const SCOPE,
+        semantics = const SEMANTICS,
+        ptr = in(reg) ptr,
+        old = in(reg) &mut old,
+        value = in(reg) &value
+    }
+
+    old
+}
+
+/// Atomically min a float and return the old value.
+#[spirv_std_macros::gpu_only]
+#[doc(alias = "OpAtomicFMinEXT")]
+#[inline]
+pub unsafe fn atomic_f_min_ext<F: Float, const SCOPE: u32, const SEMANTICS: u32>(
```

Comment:
I'm just quicky skimming over this on my phone, but I think these would probably be fine with out the `_ext` on the end of the function name. Everything else looks good though! 👍

Reply:
Okay, I removed the ext suffix 🙂